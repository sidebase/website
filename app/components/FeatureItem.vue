<script setup lang="ts">
import type { PropType } from 'vue'
import type { Feature } from '~/types'

defineProps({
  feature: {
    type: Object as PropType<Feature>,
    required: true,
  },
})
</script>

<template>
  <div class="flex flex-col space-y-4 text-center">
    <i :class="feature.icon" class="pi text-SideBase-green" style="font-size: 3rem" />
    <h1 class="text-3xl text-SideBase-green font-bold font-CooperHewitt font-light">
      {{ feature.title }}
    </h1>
    <p class="text-gray-300">
      {{ feature.description }}
    </p>
  </div>
</template>
