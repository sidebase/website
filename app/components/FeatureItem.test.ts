import { describe, expect, test } from 'vitest'
import FeatureItem from './FeatureItem.vue'
import { render } from '~/tests/testingLibraryVue'
import type { Feature } from '~/types'

const FEATURE: Feature = {
  icon: 'pi-database',
  title: 'Line 1',
  description: 'Description',
}

describe('FeatureItem', () => {
  test('mounts and is still the same', () => {
    const { html } = render(FeatureItem, { props: { feature: FEATURE } })
    expect(html()).toMatchSnapshot()
  })
})
