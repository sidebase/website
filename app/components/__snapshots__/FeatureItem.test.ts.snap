// Vitest Snapshot v1

exports[`FeatureItem > mounts and is still the same 1`] = `
"<div class=\\"flex flex-col space-y-3 text-center\\"><i class=\\"pi-database pi text-SideBase-green\\" style=\\"font-size: 3rem;\\"></i>
  <h1 class=\\"text-3xl text-SideBase-green font-bold\\">Line 1</h1>
  <p class=\\"text-gray-300\\">Description</p>
</div>"
`;
